### PR TITLE
Adds a new clown chemical 🤡

### PR DESCRIPTION
**Pull request by Citinited**
This PR adds jestosterone, which makes you behave more like a clown, and also has several slightly detrimental effects (for non-clowns)

- Effects on all mobs: Makes you HONK when hit (similar to the clown costume), makes you squeak whenever you move, makes you giggle, and makes you speak in comic sans.
- Additional effect for clowns: Heals 1.5 brute per cycle.
- Additional effects for non-clowns: Makes you see colours, makes you dizzy, and has a chance to blur your vision for a while. It will also sometimes give you warning messages (to make you see the funny side of your predicament)
- Additional effect for mimes: Take 1.5 toxin damage per cycle.

To make: 1 part banana juice, 1 part space lube, 1 part blood, 1 part space drugs, 1 part salt, heated to 374K

This PR also gives honksquads one patch of jestosterone each.

Picture:

image

:cl:
add: The chemist can now create a new clown-based reagent using banana juice, lube, blood, space drugs, and salt.
add: Jestosterone heals brute damage for clowns, triggers annoying side-effects in other crewmembers, and is toxic to mimes.
/:cl:

### CODE REVIEW

**Comment by Citinited** [comment]
The mercury is because all clowns have brain damage 😄 

I can change that to space drugs though (space drugs contain mercury anyways)

**Comment by pinatacolada** [comment]
There are way more complicated recipes for common medchems, this one is very much on the noncomplex spectrum even if you replace blood with spess drugs, i wouldnt worry about it

**Comment by Necaladun** [comment]
....is it possible to make it create a honking noise when it mixes in the heater?

**Comment by Plotron** [comment]
Space drugs are very easy to make. I'd love to see it in place of mercury, for sure.

**Comment by Fox-McCloud** [comment]
Why not just make this something you get from grinding bananium and call it...well, bananium

**Comment by Citinited** [comment]
Bananium ore is a finite resource, and I didn't want this to be finite.

I could add that as an alternative method of obtaining jestosterone though

**Comment by TDSSS** [comment]
Can you give honksquads syringe guns loaded with it, for making the entire station more jolly?

**Comment by Citinited** [comment]
They already have snap-pop rifles for that, though I could feasibly add it. I'll play around with it tonight

**Comment by Citinited** [comment]
No, it's determined by the mind's assigned role. I believe this is how similar checks are done elsewhere in the code

**Comment by Fox-McCloud** [comment]
I admit, I'm kinda skeptical of having every single say message and every single movement by a carbon on up checking for a specific reagent.

**Comment by Citinited** [comment]
I'll rework this to use components - I just need to learn how to use them first 😄

**Comment by Fox-McCloud** [comment]
Do you want that to be in this PR, or should I close this until you get that handled?

**Comment by Citinited** [comment]
@Fox-McCloud Yeah, it'll be in this PR, no need to close it

**Comment by Citinited** [comment]
@Fox-McCloud I've added a jestosterone component, let me know of any code improvements you can see

**Comment by Citinited** [comment]
(sorry f

**Comment by Citinited** [comment]
(sorry for double post) I'll add mime-specific stuff later, I mostly uploaded this for component feedback

**Comment by Fox-McCloud** [comment]
https://github.com/tgstation/tgstation/blob/master/code/datums/components/squeak.dm

This is a more thorough implementation.

File: code/datums/components/jestosterone.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	mob_is_clown = is_this_mob_a_clown
+	if(mob_is_clown)
+		to_chat(user, "<span class='notice'>Whatever that was, it feels great!</span>")
+	else
+		to_chat(user, "<span class='warning'>Something doesn't feel right...</span>")
+	RegisterSignal(COMSIG_MOVABLE_MOVED, .proc/on_move)
+	RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/mob_was_attacked)
+	RegisterSignal(COMSIG_ITEM_ATTACK, .proc/mob_was_attacked)
+	RegisterSignal(COMSIG_ATOM_ATTACK_ALIEN, .proc/mob_was_attacked)
+	RegisterSignal(COMSIG_ON_MOB_LIFE, .proc/process_jestosterone)
```
Comment (Crazylemon64) [inline code/datums/components/jestosterone.dm]: wait what

this should be in `on_mob_life` of the reagent in question.

Also, I'd advocate porting the `redirect` component, and putting these inside the `on_new` of the `jestosterone` reagent.

Then, in `on_mob_delete`, remove the redirect components from the mob.

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 2]
```diff
@@ -421,6 +421,32 @@
...
+		return
+	var/is_a_clown = FALSE
+	if(M.mind && (M.mind.assigned_role == "Clown" || M.mind.assigned_role == SPECIAL_ROLE_HONKSQUAD))
+		is_a_clown = TRUE
+	if(!is_a_clown)
+		M.AdjustDizzy(volume)
+	M.AddComponent(/datum/component/jestosterone, M, is_a_clown)
+
+/datum/reagent/jestosterone/on_mob_life(mob/living/M)
+	M.SendSignal(COMSIG_ON_MOB_LIFE, M)
```
Comment (Crazylemon64) [inline code/modules/reagents/chemistry/reagents/misc.dm]: :eyes:

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 3]
```diff
@@ -421,6 +421,32 @@
...
+/datum/reagent/jestosterone/reaction_mob(mob/living/M, method, volume)
+	..()
+	if(istype(M, /mob/living/carbon))
+		return
+	var/is_a_clown = FALSE
+	if(M.mind && (M.mind.assigned_role == "Clown" || M.mind.assigned_role == SPECIAL_ROLE_HONKSQUAD))
+		is_a_clown = TRUE
+	if(!is_a_clown)
+		M.AdjustDizzy(volume)
+	M.AddComponent(/datum/component/jestosterone, M, is_a_clown)
```
Comment (Crazylemon64) [inline code/modules/reagents/chemistry/reagents/misc.dm]: With `reaction_mob`, if this reagent were applied in a manner which would not add it to the list of the mob's reagents, such as a beaker splash, this component would be added and never removed. Use `on_new` instead.

File: code/modules/reagents/chemistry/reagents/misc.dm (hunk 2 above)
Comment (Crazylemon64) [reply code/modules/reagents/chemistry/reagents/misc.dm]: (if you *were* to implement this, this should be a general component, not individually fired on a single reagent. but reagents already have a method of doing stuff on mob life, so use that instead)

**Comment by Crazylemon64** [comment]
Are you going to update this PR?

**Comment by Citinited** [comment]
@Crazylemon64 I've been putting this off, I know (you have Monster Hunter to thank XD). I'll definitely update it tomorrow though 😄

File: code/modules/reagents/chemistry/reagents/misc.dm (hunk 3 above)
Comment (Citinited) [reply code/modules/reagents/chemistry/reagents/misc.dm]: I may be missing something, but `on_new` doesn't seem to be called anywhere relevant - I tested by doing this:
```
/datum/reagent/proc/on_new(data)
	to_chat(world, "new reagent")
	return
```
And `new reagent` was never printed for me.

Should I use `New` instead?

File: code/modules/reagents/chemistry/reagents/misc.dm (hunk 2 above)
Comment (Citinited) [reply code/modules/reagents/chemistry/reagents/misc.dm]: Done

**Comment by Citinited** [comment]
I'm going to close this as there isn't really any time soon I can look at it to change some of the things I'd like to. It will be back at a later date:tm:

**Comment by Citinited** [comment]
Reopening this! Now that the squeak component has been ported it should be relatively easy to finish this.

**Comment by Citinited** [comment]
Committed more changes:
- Worked this in to use the new squeak component
- Fixes an issue with reagents not getting `on_new` called
- jestosterone is now toxic to mimes

**Comment by Diegoflores31** [comment]
We can, give it some you know like a danger , when heated makes Honk noise and strips your shoes out ? Like the H.O.N.k blast

**Comment by Citinited** [comment]
I think the effects on mobs are annoying enough for this 😄

**Pull request merged**